langchain/llms/mlflow.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+from functools import lru_cache
+from typing import Any, List, Dict, Mapping, Optional
+from langchain.callbacks.manager import CallbackManagerForLLMRun
+from langchain.llms.base import LLM
+from pydantic import BaseModel, Extra
+from typing import TYPE_CHECKING
+
+
+if TYPE_CHECKING:
+    import mlflow.gateway
+
+
+@lru_cache()
+def _get_client(gateway_uri: str) -> mlflow.gateway.MlflowGatewayClient:
+    import mlflow.gateway
+
+    return mlflow.gateway.MlflowGatewayClient(gateway_uri)
```

Comment:
Tried `self.client = mlflow.gateway.MlflowGatewayClient(...)` first, but rejected by pydantic, then switched to this approach.